packages/std-client/src/setup/setupMUDV2Network.ts

```diff
@@ -0,0 +1,186 @@
+import { MUDUserConfig } from "@latticexyz/cli";
+
+type SetupMUDV2NetworkOptions<M extends MUDUserConfig = MUDUserConfig> = {
+  mudConfig: M;
+  networkConfig: SetupContractConfig;
+  initialGasPrice?: number;
+  fetchSystemCalls?: boolean;
+  syncThread?: "main" | "worker";
+};
+
+export async function setupMUDV2Network<M extends MUDUserConfig = MUDUserConfig>({
+  mudConfig,
+  networkConfig,
+  initialGasPrice,
+  fetchSystemCalls,
+  syncThread,
+}: SetupMUDV2NetworkOptions<M>) {
+  const world = createWorld();
+
+  const SystemsRegistry = defineStringComponent(world, {
+    id: "SystemsRegistry",
+    metadata: { contractId: "world.component.systems" },
+  });
+  const ComponentsRegistry = defineStringComponent(world, {
+    id: "ComponentsRegistry",
+    metadata: { contractId: "world.component.components" },
+  });
+
+  // used by SyncWorker to notify client of sync progress
+  const LoadingState = defineComponent(
+    world,
+    {
+      state: Type.Number,
+      msg: Type.String,
+      percentage: Type.Number,
+    },
+    {
+      id: "LoadingState",
+      metadata: { contractId: "component.LoadingState" },
+    }
+  );
+
+  const storeSchemaTableId = new TableId("mudstore", "schema");
+  const storeSchemaComponent = defineComponent(
+    world,
+    { schema: Type.T },
+    {
+      metadata: {
+        contractId: storeSchemaTableId.toHexString(),
+        tableId: storeSchemaTableId.toString(),
+      },
+    }
+  );
+
+  const storeComponents = defineStoreComponents(world, storeMudConfig);
+  const worldComponents = defineStoreComponents(world, worldMudConfig);
+  const userComponents = defineStoreComponents(world, mudConfig);
+
+  const components = {
+    // v2 components
+    storeSchemaComponent,
+    ...storeComponents,
+    ...worldComponents,
+    ...userComponents,
+    // v1 components
+    SystemsRegistry,
+    ComponentsRegistry,
+    LoadingState,
+  } satisfies Components;
+
+  // Mapping from component contract id to key in components object
+  const mappings: Mappings<typeof components> = {};
+
+  // Function to register new components in mappings object
+  function registerComponent(key: string, component: ContractComponent) {
+    const { contractId, tableId } = component.metadata;
+    if (tableId) {
+      mappings[tableId] = key;
+    } else {
+      mappings[keccak256(contractId)] = key;
+    }
+  }
+
+  // Register initial components in mappings object
+  for (const key of Object.keys(components)) {
+    registerComponent(key, components[key]);
+  }
+
+  const network = await createNetwork(networkConfig);
+  world.registerDisposer(network.dispose);
+
+  const signerOrProvider = computed(() => network.signer.get() || network.providers.get().json);
+
+  const { contracts, config: contractsConfig } = await createContracts<{ World: WorldContract }>({
+    config: { World: { abi: WorldAbi, address: networkConfig.worldAddress } },
+    signerOrProvider,
+  });
```

Comment:
In a v2 environment the provided worldAddress won't correspond to the `solecs` `WorldAbi`. I'm happy with still merging this though and iterating on it, and eventually replacing the network package with a fresh sync package for v2.

Reply:
Would it be a bad idea to use the v2world ABI/contract here? Not sure what the txQueue and other downstream stuff depends on.

<img width="836" alt="image" src="https://user-images.githubusercontent.com/508855/228571351-54d34783-8e41-4f11-9a8a-5264633249db.png">

Reply:
Unfortunately the interface from `world` doesn't include the user defined function selectors (which are generated by `mud worldgen && mud worldtypes` in the project repo)

Reply:
we can require users to pass their world interface as an argument though, similar to before they had to pass the system types object

Reply:
Since both this setup function and worldgen depend on the mud config, I wonder if we have enough info to avoid having to pass through the artifact of worldgen into this?

Ideally, our mud config approach looks something like:
- user supplies simplified config (with optional shorthands, etc.)
- method + TS types that "expand" the config into something more verbose and useful for downstream things like codegen
  - the result of this would be used to populate things like component definitions and types, etc.
- the expanded config is used by things like worldgen, etc.

The only thing that breaks this is that system methods (and thus the world interface) are based on the filesystem rather than config.

Reply:
#536 reimagines how setup works, so if we go with tsgen it can do all the work and setup just gets readymade tables+systems